strapi/strapi
# feat(blocks): add markdown snippets

### What does it do?

Adds some snippets that convert the current block into a different one. They come from markdown:

- `#` to `######` to convert to a heading
- `>` to convert to a quote
- `-`, `*` or `+` to convert to a bullet list
- `1.` to convert to a numbered list
- `![` to convert to an image
- three times ` to convert to a code block

They get triggered when pressing the spacebar, only if the selection is a the start of the block with nothing after it.

I also extracted the logic to render the image modal returned by handleConvert into a hook. It's now used for the toolbar selector and the snippets, and soon will be used for slash commands too.

Unfortunately no tests as the feature relies on event listeners on a content editable dom node, [which is not supported by RTL](https://stackoverflow.com/a/77054841/3661792). We need E2E tests for this, which I think is its own task

### Why is it needed?

Requested by users

### How to test it?

Type the snippets in the editor and see the transformations. Also try to type them at the end of a line, or at the beginning with content after. Try them in the middle of a list too

### Related issue(s)/PR(s)

made possible by #18840
